src/app/components/quiz-page/quiz-page.component.ts

```diff
@@ -1,58 +1,96 @@
-import {Component, OnInit} from '@angular/core';
+import {Component, OnDestroy, OnInit} from '@angular/core';
 import {QuizDefinitionModel} from 'src/app/models/quiz-definition-model';
 
 import {AnswerDto} from '../../models/answer-dto';
 import {AnswerModel} from '../../models/answer-model';
-import {QuizService} from "../../service/quiz.service";
-import {Router} from "@angular/router";
-
+import {QuizService} from '../../service/quiz.service';
+import {Router} from '@angular/router';
+import {interval, Subscription} from 'rxjs';
 
 @Component({
   selector: 'app-quiz-page',
   templateUrl: './quiz-page.component.html',
   styleUrls: ['./quiz-page.component.scss']
 })
-export class QuizPageComponent implements OnInit {
+export class QuizPageComponent implements OnInit, OnDestroy {
 
   SESSION_STORAGE_USER_ID = 'BILLENNIUM_CANDIDATE_ID';
-
+  DEFAULT_TIMER = -1;
+  SESSION_STORAGE_TIMER = 'CANDIDATE_TIMER';
+  SESSION_STORAGE_QUIZ = 'CANDIDATE_QUIZ';
+  ANSWER_TIME_OUT = 0;
   response: QuizDefinitionModel;
   selectedAnswer = new AnswerModel();
   request = new AnswerDto();
-
+  timeToAnswer = -1;
+  timer: Subscription;
 
   constructor(private quizService: QuizService,
               private router: Router) {
   }
 
+
   ngOnInit(): void {
-    this.prepareInitRequestBody();
-    this.quizService.saveAndLoadQuestions(sessionStorage.getItem(this.SESSION_STORAGE_USER_ID), this.request)
-      .subscribe(data => {
-        this.response = data;
-      });
+    this.timeToAnswer = Number(sessionStorage.getItem(this.SESSION_STORAGE_TIMER));
+
+    if (sessionStorage.getItem(this.SESSION_STORAGE_QUIZ) === '') {
+      this.sendData(null, null);
```

Comment:
musi to tak byc ? czyli nie masz tego tokena i wysylasz z nullami ?

Reply:
te nulle sa wrzucane do body bo przyejslismy ze w chwili wyslania zpaytania o pierwsze pytanie wyslamy id pytania i id opdoiwedzi jako null